### PR TITLE
Refactor/fix ivanalysis

### PR DESCRIPTION
**Pull request by cwfink**
Updated the `IVAnalysis` class to be compatible with the new `IBIS` and `DIDV` classes in `QETpy`. This PR also adds all the functionality to the `IVAnalysis` class that was in the branch https://github.com/ucbpylegroup/RQpy/tree/refactor/IVerror_prop 

I've re-analyzed SLAC PD2 data with this updated code and confirmed that all the functionality works.

### CODE REVIEW

File: rqpy/plotting/_iv_didv_tools_plotting.py [hunk 1]
```diff
@@ -257,3 +323,455 @@ def _plot_energy_res_vs_bias(r0s,
...
+            np.sqrt(np.abs(data.noise_model['ptot'][2][ii])),
+            alpha=.5,
+            color='#d62728',
+        )
+        ax.loglog(
+            freqs,
+            np.sqrt(np.abs(data.noise_model['psquid'][0][ii])),
+            color='#9467bd',
+            linewidth=1.5,
+            label='Squid+Electronics Noise',
```
Comment (slwatkins) [inline rqpy/plotting/_iv_didv_tools_plotting.py:750]: Perhaps we should replace the `+` with an ampersand `&`? I feel that the plus sign is a little colloquial for plots.
```suggestion
            label='Squid & Electronics Noise',
```

File: rqpy/plotting/_iv_didv_tools_plotting.py [hunk 2]
```diff
@@ -257,3 +323,455 @@ def _plot_energy_res_vs_bias(r0s,
...
+            np.sqrt(np.abs(data.noise_model['itot'][2][ii])),
+            alpha=.5,
+            color='#d62728',
+        )
+        ax.loglog(
+            freqs,
+            np.sqrt(np.abs(data.noise_model['isquid'][0][ii])),
+            color='#9467bd',
+            linewidth=1.5,
+            label='Squid+Electronics Noise',
```
Comment (slwatkins) [inline rqpy/plotting/_iv_didv_tools_plotting.py:654]: Perhaps we should replace the `+` with an ampersand `&`? I feel that the plus sign is a little colloquial for plots.
```suggestion
            label='Squid & Electronics Noise',
```

File: rqpy/plotting/_iv_didv_tools_plotting.py [hunk 3]
```diff
@@ -103,121 +139,146 @@ def _plot_rload_rn_qetbias(IVanalysisOBJ, lgcsave, xlims_rl, ylims_rl, xlims_rn,
...
+        IVanalysisOBJ.vb[0,0,IVanalysisOBJ.norminds]*1e6,
+        np.array(IVanalysisOBJ.rtot_list)*1e3,
+        yerr=IVanalysisOBJ.rshunt_err*1e3,
+        linestyle='',
+        marker='.',
+        ms=10,
+    )
+    axes[1].grid(True, linestyle='dashed')
+    axes[1].set_title('Rtotal vs Vbias', fontsize=14)
+    axes[1].set_ylabel(r'$R_{N} + R_ℓ$ [mΩ]', fontsize=14)
```
Comment (slwatkins) [inline rqpy/plotting/_iv_didv_tools_plotting.py:168]: Does this labeling work with the ℓ and Ω characters? Or should we switch to their LaTeX versions (\ell, \Omega)

```suggestion
    axes[1].set_ylabel(r'$R_{N} + R_\ell$ [m$\Omega$]', fontsize=14)
```

Comment (slwatkins) [reply rqpy/plotting/_iv_didv_tools_plotting.py:168]: I know that Python 3 should support it, but sometimes the LaTeX compiler errors on the special characters in between $'s

File: rqpy/plotting/_iv_didv_tools_plotting.py (hunk 1 above)
Comment (cwfink) [reply rqpy/plotting/_iv_didv_tools_plotting.py]: added

File: rqpy/plotting/_iv_didv_tools_plotting.py (hunk 2 above)
Comment (cwfink) [reply rqpy/plotting/_iv_didv_tools_plotting.py]: added

File: rqpy/plotting/_iv_didv_tools_plotting.py (hunk 3 above)
Comment (cwfink) [reply rqpy/plotting/_iv_didv_tools_plotting.py]: changed this, and a few more places as well

File: rqpy/plotting/_iv_didv_tools_plotting.py [hunk 4]
```diff
@@ -149,8 +149,8 @@ def _plot_rload_rn_qetbias(IVanalysisOBJ, lgcsave, xlims_rl, ylims_rl,
     )
     axes[0].grid(True, linestyle = 'dashed')
     axes[0].set_title('Rload vs Vbias', fontsize = 14)
-    axes[0].set_ylabel(r'$R_ℓ$ [mΩ]', fontsize = 14)
-    axes[0].set_xlabel(r'$V_{bias}$ [μV]', fontsize = 14)
+    axes[0].set_ylabel(r'$R_\ell \, [\mathrm{m}\Omega]$', fontsize = 14)
+    axes[0].set_xlabel(r'$V_\text{bias} \, [\mu\mathrm{V}]$', fontsize = 14)
```
Comment (slwatkins) [inline rqpy/plotting/_iv_didv_tools_plotting.py:153]: The `\text` function I don't think is supported in matplotlib LaTeX
```suggestion
    axes[0].set_xlabel(r'$V_\mathrm{bias} \, [\mu\mathrm{V}]$', fontsize = 14)
```

File: rqpy/plotting/_iv_didv_tools_plotting.py [hunk 5]
```diff
@@ -165,8 +165,8 @@ def _plot_rload_rn_qetbias(IVanalysisOBJ, lgcsave, xlims_rl, ylims_rl,
     )
     axes[1].grid(True, linestyle='dashed')
     axes[1].set_title('Rtotal vs Vbias', fontsize=14)
-    axes[1].set_ylabel(r'$R_{N} + R_ℓ$ [mΩ]', fontsize=14)
-    axes[1].set_xlabel(r'$V_{bias}$ [μV]', fontsize=14)
+    axes[1].set_ylabel(r'$R_N + R_\ell \, [\mathrm{m}\Omega]$', fontsize=14)
+    axes[1].set_xlabel(r'$V_\text{bias} \, [\mu\mathrm{V}]$', fontsize=14)
```
Comment (slwatkins) [inline rqpy/plotting/_iv_didv_tools_plotting.py:169]: The `\text` function I don't think is supported in matplotlib LaTeX
```suggestion
    axes[1].set_xlabel(r'$V_\mathrm{bias} \, [\mu\mathrm{V}]$', fontsize=14)
```

File: rqpy/plotting/_iv_didv_tools_plotting.py [hunk 6]
```diff
@@ -1,57 +1,91 @@
...
+    "_plot_ztes_bias",
+    "_plot_noise_model",
+]
 
 
 def _make_iv_noiseplots(IVanalysisOBJ, lgcsave=False):
     """
-    Helper function to plot average noise/didv traces in time domain, as well as 
-    corresponding noise PSDs, for all QET bias points in IV/dIdV sweep.
+
```
Comment (slwatkins) [inline rqpy/plotting/_iv_didv_tools_plotting.py:21]: Whoops, I guess I added an extra return here!

```suggestion
```

**Review by slwatkins — APPROVED** [review]
After the small fixes, I haven't noticed any glaring issues. @cwfink I'll approve this for now - do you want to do one last analysis sweep check before merging?

**Comment by cwfink** [comment]
Found syntax error. Doing one final analysis sweep check, then will merge once everything checks out

**Comment by cwfink** [comment]
Found and fixed an issue with how the priors covariance matrix was constructed when doing the 3 pole priors fit.

File: rqpy/core/_iv_didv_tools.py [hunk 7]
```diff
@@ -1389,6 +1397,7 @@ def estimate_noise_errors(self, tau_collect=0, collection_eff=1,
             tes_params = self._get_tes_params(
                 didvobj, nsamples=nsamples,
             )
+            print(ind)
```
Comment (slwatkins) [inline rqpy/core/_iv_didv_tools.py:1400]:
```suggestion
            
```

Extra print statement?

**Pull request merged**